plugins/askrene/layer.c

```diff
@@ -870,12 +1008,53 @@ const struct bias *layer_set_bias(struct layer *layer,
 	return bias;
 }
 
+const struct node_bias *layer_set_node_bias(struct layer *layer,
+					    const struct node_id *node,
+					    const char *description TAKES,
+					    s8 bias_factor,
+					    bool relative,
+					    bool dir_out,
+					    u64 timestamp)
+{
+	const struct node_bias *bias;
+
+	bias = set_node_bias(layer, node, description, bias_factor, relative,
+			     dir_out, timestamp);
+	save_node_bias(layer, bias);
+	return bias;
+}
+
 void layer_apply_biases(const struct layer *layer,
 			const struct gossmap *gossmap,
 			s8 *biases)
 {
 	struct bias *bias;
 	struct bias_hash_iter it;
+	struct node_bias *node_bias;
+	struct node_bias_hash_iter node_it;
+
+	/* We assume bias from individual channels and their node are additive,
+	 * this is completely arbitrary. */
+	for (node_bias = node_bias_hash_first(layer->node_biases, &node_it);
+	     node_bias;
+	     node_bias = node_bias_hash_next(layer->node_biases, &node_it)) {
+		struct gossmap_node *n;
+		struct gossmap_chan *c;
+		int dir;
+		u32 idx;
+
+		n = gossmap_find_node(gossmap, &node_bias->node);
+		if (!n)
+			continue;
+		for (size_t i = 0; i < n->num_chans; i++) {
+			c = gossmap_nth_chan(gossmap, n, i, &dir);
+			idx = (gossmap_chan_idx(gossmap, c) << 1) | dir;
+
+			// FIXME: is this the correct direction?
+			biases[idx] += node_bias->out_bias;
+			biases[idx ^ 1] += node_bias->in_bias;
```

Comment:
To answer your FIXME: yes, it's correct.

The direction for a node pair corresponds to the outgoing channel: that's because the node controls fees only on outgoing.  So, biases[index] is indeed this node's outgoing.